src/MagnumPlugins/BasisImporter/Test/BasisImporterTest.cpp

```diff
@@ -0,0 +1,246 @@
+        false
+    }, {
+        "Bc1", "Bc1",
+        CompressedPixelFormat::Bc1RGBUnorm,
+        CompressedPixelFormat::Bc1RGBAUnorm,
+        false
+    }, {
+        "Bc3", "Bc3",
+        CompressedPixelFormat::Bc3RGBAUnorm,
+        CompressedPixelFormat::Bc3RGBAUnorm,
+        false
+    }, {
+        "Bc4", "Bc4",
+        CompressedPixelFormat::Bc4RUnorm,
+        CompressedPixelFormat::Bc4RUnorm,
+        false
+    }, {
+        "Bc5", "Bc5",
+        CompressedPixelFormat::Bc5RGUnorm,
+        CompressedPixelFormat::Bc5RGUnorm,
+        false
+    }, {
+        "Bc7M6OpaqueOnly", "Bc7M6OpaqueOnly",
+        CompressedPixelFormat::Bc7RGBAUnorm,
+        CompressedPixelFormat::Bc7RGBAUnorm,
+        false
+    }, {
+        "Pvrtc1_4OpaqueOnly", "Pvrtc1_4OpaqueOnly",
+        CompressedPixelFormat::PvrtcRGB4bppUnorm,
+        CompressedPixelFormat::PvrtcRGB4bppUnorm,
+        true
+    }
+};
+
+BasisImporterTest::BasisImporterTest() {
+    addTests({&BasisImporterTest::empty,
+              &BasisImporterTest::invalid,
+              &BasisImporterTest::unconfigured,
+
+              &BasisImporterTest::openTwice,
+              &BasisImporterTest::importTwice});
+
+    addInstancedTests({&BasisImporterTest::rgb,
+                       &BasisImporterTest::rgba},
+                       Containers::arraySize(FormatData));
+
+    /* Load the plugin directly from the build tree. Otherwise it's static and
+       already loaded. */
+    #ifdef BASISIMPORTER_PLUGIN_FILENAME
+    CORRADE_INTERNAL_ASSERT(_manager.load(BASISIMPORTER_PLUGIN_FILENAME) & PluginManager::LoadState::Loaded);
+    #endif
+}
+
+void BasisImporterTest::empty() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    importer->configuration().setValue("format", "Etc2");
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    char a{};
+    /* Explicitly checking non-null but empty view */
+    CORRADE_VERIFY(!importer->openData({&a, 0}));
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::openData(): the file is empty\n");
+}
+
+void BasisImporterTest::invalid() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!importer->openData("NotABasisFile"));
+
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::openData(): invalid header\n");
+}
+
+void BasisImporterTest::unconfigured() {
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporter");
+    std::ostringstream out;
+    Error redirectError{&out};
+
+    CORRADE_VERIFY(importer->openFile(
+        Utility::Directory::join(BASISIMPORTER_TEST_DIR, "rgb.basis")));
+    CORRADE_VERIFY(!importer->image2D(0));
+
+    CORRADE_COMPARE(out.str(), "Trade::BasisImporter::image2D(): invalid configuration: format is expected to be one of: Etc1, Etc2, Bc1, Bc3, Bc4, Bc5, Bc7M6OpaqueOnly, Pvrtc1_4OpaqueOnly\n");
```

Comment:
This is exactly what I meant by "have an explicit test so we are sure this fails gracefully with a good UX" ;) This error would be valid if you set a format to some bogus value, but then it should say *what* the wrong format is -- so if you set `format=Bananas`, it complains with

> Trade::BasisImporter::image2D(): invalid configured transcoding format Bananas, expected to be one of [...]

However, when the format is empty (which is the default case), from the user POV the above doesn't really say *what to do*. In that case, something like this would be better:

> Trade::BasisImporter::image2D(): no format to transcode to was specified. Either load the plugin via one of its BasisImporterEtc1, ... aliases, or set the format explicitly via plugin configuration.

This also means there should be two tests for this, one `unconfigured()` and one `wrongConfiguredFormat()`.